runtime/codert_vm/decomp.cpp

```diff
@@ -1499,42 +1619,39 @@ decompileAllFrameIterator(J9VMThread * currentThread, J9StackWalkState * walkSta
 {
 	/* Decompile all JIT frames */
 
-	if (walkState->jitInfo) {
-		addDecompilation(currentThread, walkState, (UDATA) walkState->userData1);
+	if (NULL != walkState->jitInfo) {
+#if JAVA_SPEC_VERSION >= 19
+		J9VMContinuation *continuation = (J9VMContinuation*)walkState->userData2;
+		if (NULL != continuation) {
+			addDecompilationForContinuation(currentThread, walkState, (UDATA) walkState->userData1, continuation)	;
+		} else
+#endif /* JAVA_SPEC_VERSION >= 19 */
+		{
+			addDecompilation(currentThread, walkState, (UDATA) walkState->userData1);
+		}
 	}
 
 	return J9_STACKWALK_KEEP_ITERATING;
 }
 
 
 static void
-deleteAllDecompilations(J9VMThread * currentThread, UDATA reason, J9Method * method)
+deleteAllDecompilations(J9VMThread *currentThread, UDATA reason, J9Method * method)
 {
-	J9VMThread * loopThread;
+	J9VMThread * loopThread = currentThread;
 
 	Trc_Decomp_deleteAllDecompilations_Entry(currentThread);
 
-	loopThread = currentThread;
 	do {
-		J9JITDecompilationInfo ** previous;
-		J9JITDecompilationInfo * current;
-
-		previous = &(loopThread->decompilationStack);
-		while ((current = *previous) != NULL) {
-			if ((current->reason & reason) && (!method || (current->method == method))) {
-				current->reason &= ~reason;
-				if (current->reason) {
-					Trc_Decomp_deleteAllDecompilations_notFreeingRecord(currentThread, current, current->reason);
-					previous = &(current->next);
-				} else {
-					*previous = deleteDecompilationForExistingFrame(loopThread, current);
-				}
-			} else {
-				previous = &(current->next);
-			}
-		}
+		freeDecompilations(currentThread, loopThread, &loopThread->decompilationStack, reason, method);
 	} while ((loopThread = loopThread->linkNext) != currentThread);	
 
+#if JAVA_SPEC_VERSION >= 19
+	J9JavaVM *vm = currentThread->javaVM;
+	J9DeleteDecompData data = { reason, method };
+	vm->memoryManagerFunctions->j9mm_iterate_all_continuation_objects(currentThread, vm->portLibrary, 0, deleteDecompCallBack, (void*)&data);
```

Comment:
Does invoking `j9gc_flush_nonAllocationCaches_for_walk` from `jvmtiBreakpoint.c` support all `j9mm_iterate_all_continuation_objects` calls? There will be better flow and easier verification if `j9gc_flush_nonAllocationCaches_for_walk` and `j9mm_iterate_all_continuation_objects` are invoked together.

Reply:
If that's the case, then the GC should just do it in the iterate call. Nothing will be allocated after the flush call, so it should not be necessary to call it for every continuation.

Reply:
If nothing will be allocated between the flush and iterate_all due to exclusive VM access, then the current impl is fine. We just need a comment to document this behaviour.

Reply:
This is implied by having exclusive.